torchio/transforms/transform.py

```diff
@@ -51,7 +51,7 @@ def __call__(self, data: Union[Subject, torch.Tensor]):
             sample = data
         self.parse_sample(sample)
         if torch.rand(1).item() > self.probability:
-            return sample
+            return data
```

Comment:
What about just moving these two lines to the beginning of `__call__`? Then there would be no need to parse the input.

Reply:
I had initially done it like that, however in that case input type errors would appear only with a certain probability making debugging harder. On the other side the parsing needlessly slows the computation...

Reply:
Yeah, that's a good point. But for debugging it wouldn't make sense to use probability less than 1 anyway, and I'd like to start optimizing things, the library can be a bit slow sometimes.

Reply:
Alright, I will make the change asap.